src/openms/source/FORMAT/HANDLERS/XMLHandler.cpp

```diff
@@ -82,15 +82,18 @@ namespace OpenMS::Internal
       if (mode == LOAD)
       {
         error_message =  String("While loading '") + file_ + "': " + msg;
-	      // test if file has the wrong extension and is therefore passed to the wrong parser
-        // only makes sense if we are loading/parsing a file
-	      FileTypes::Type ft_name = FileHandler::getTypeByFileName(file_);
-        FileTypes::Type ft_content = FileHandler::getTypeByContent(file_);
-        if (ft_name != ft_content)
+        if (!file_.hasPrefix("http://") && !file_.hasPrefix("https://") && !file_.hasPrefix("ftp://") && !file_.hasPrefix("s3://"))
```

Comment:
this kind of query looks useful in a more general scope.
Since its used repetitively here, can this the externalized to something like:

```cpp
  /// check if filename starts with http, https, ftp or s3 followed by `://`
  bool File::isRemoteFile(const std::string& filename);
```